### PR TITLE
Update network documentation

### PR DESCRIPTION
**Pull request by sgrasley**
Adding documentation about tunnel network, external network and
adding note about vmware network requirement.

### CODE REVIEW

File: doc/source/deployment/requirements.rst [hunk 1]
```diff
@@ -94,6 +94,20 @@ Network Requirements
     separate network should be present on the Caas Platform and ses_config.yml
     mon_host: section.
 
+* Tunnel Network
+    A network defined by its network interface. It must be up and have ip
+    address defined. This is used for tunneling vxlan traffic for tennant vm's
```
Comment (csymons-suse) [inline doc/source/deployment/requirements.rst]: an IP address defined. ...tunneling VXLAN traffic for tenant VMs

File: doc/source/deployment/requirements.rst [hunk 2]
```diff
@@ -94,6 +94,20 @@ Network Requirements
...
+    within the cloud.
+
+* External Network
+    A network defined by its network interface. This interface should not have
+    an IP address defined (will not be accessable when cloud deployed). The
+    network interface should be configured BOOTPROTO=none and active.
+
+  .. note::
+     If installing on vmware infrastructure, make sure this network has
+     security set to allow permiscuous mode and forged transmit.
```
Comment (csymons-suse) [inline doc/source/deployment/requirements.rst]: promiscuous

File: doc/source/deployment/requirements.rst [hunk 3]
```diff
@@ -94,6 +94,20 @@ Network Requirements
...
+    address defined. This is used for tunneling vxlan traffic for tennant vm's
+    within the cloud.
+
+* External Network
+    A network defined by its network interface. This interface should not have
+    an IP address defined (will not be accessable when cloud deployed). The
+    network interface should be configured BOOTPROTO=none and active.
+
+  .. note::
+     If installing on vmware infrastructure, make sure this network has
```
Comment (csymons-suse) [inline doc/source/deployment/requirements.rst]: VMware

File: doc/source/deployment/requirements.rst [hunk 4]
```diff
@@ -94,6 +94,20 @@ Network Requirements
...
 
+* Tunnel Network
+    A network defined by its network interface. It must be up and have ip
+    address defined. This is used for tunneling vxlan traffic for tennant vm's
+    within the cloud.
+
+* External Network
+    A network defined by its network interface. This interface should not have
+    an IP address defined (will not be accessable when cloud deployed). The
+    network interface should be configured BOOTPROTO=none and active.
```
Comment (csymons-suse) [inline doc/source/deployment/requirements.rst]: accessible
should be configured `BOOTPROTO=none`

File: doc/source/deployment/requirements.rst (hunk 2 above)
Comment (sgrasley) [reply doc/source/deployment/requirements.rst]: fixed

File: doc/source/deployment/requirements.rst (hunk 4 above)
Comment (sgrasley) [reply doc/source/deployment/requirements.rst]: fixed

File: doc/source/deployment/requirements.rst (hunk 3 above)
Comment (sgrasley) [reply doc/source/deployment/requirements.rst]: fixed

**Pull request merged**